concordances/cypher.go

```
@@ -48,25 +49,151 @@ type neoResultStrunct struct {
-		RETURN collect({uuid:p.uuid, types:labels(p), neoIdentifier:{labels:labels(ids), value:ids.value}}) as rs
+		WHERE NOT ids:UPPIdentifier
+		RETURN p.uuid as prefUUID, p.uuid AS UUID, labels(p) AS TYPES, {labels:labels(ids), value:ids.value} as IDENTIFIERS
 		`,
 		Parameters: neoism.Props{"identifiers": identifiers},
 		Result:     &results,
 	}
-
 	return processCypherQueryToConcordances(pcw, query, &results)
 }
 
 func (pcw CypherDriver) ReadByAuthority(authority string, identifierValues []string) (concordances Concordances, found bool, err error) {
+	c, f, err := pcw.readByAuthorityNewModel(authority, identifierValues)
+	if !f {
+		c, f, err = pcw.readByAuthorityOldModel(authority, identifierValues)
+	}
+	return c, f, err
+}
+
+func (pcw CypherDriver) readByAuthorityNewModel(authority string, identifierValues []string) (concordances Concordances, found bool, err error) {
+	log.Debug("readByAuthorityNewModel")
+	concordances = Concordances{}
+	results := []neoReadStruct{}
+
+	authorityProperty := mapAuthorityToAuthorityProperty(authority)
+	log.Info(authorityProperty)
+	if authorityProperty == "" {
+		return Concordances{}, false, nil
+	}
+
+	readByAuthorityQueryStatement := `
+		MATCH (p:Concept)-[:EQUIVALENT_TO]-(cn:Concept)
+		WHERE p.authorityValue in {identifierValues} AND p.authority = {authority}
+		RETURN cn.prefUUID as prefUUID, cn.prefUUID AS UUID, labels(cn) AS TYPES, {labels:collect(p.authority), value:p.authorityValue} as IDENTIFIERS
+		`
+
+	query := &neoism.CypherQuery{
+		Statement: readByAuthorityQueryStatement,
+		Parameters: neoism.Props{
+			"identifierValues": identifierValues,
+			"authority":        authorityProperty,
+		},
+		Result: &results,
+	}
+
+	err = pcw.conn.CypherBatch([]*neoism.CypherQuery{query})
+	if err != nil {
+		log.Errorf("Error looking up Concordances with query %s from neoism: %+v\n", query.Statement, err)
+		return Concordances{}, false, fmt.Errorf("Error accessing Concordance datastore for identifier:")
```

shouldnt we return the actual error?